File: apps/desktop/src/renderer/pages/toolbox/market/skillStoreGroups.ts
import type { SkillStoreGroup } from "@/services/skillStore";
import { isSkillStoreGroup } from "@/services/skillStore";

export const SKILL_STORE_GROUPS = [
  { id: "legal", label: "法律合规" },
  { id: "writing", label: "写作成稿" },
  { id: "research", label: "研究调研" },
  { id: "product", label: "产品商业" },
  { id: "engineering", label: "开发工程" },
  { id: "decision", label: "决策对抗" },
] as const satisfies readonly { id: SkillStoreGroup; label: string }[];

export function skillStoreGroupLabel(id: SkillStoreGroup): string {
  return SKILL_STORE_GROUPS.find((row) => row.id === id)?.label ?? id;
}

export { isSkillStoreGroup };
export type { SkillStoreGroup };

File: apps/server/agentcore/db/skill_store_groups.py
"""Closed scene groups for the Skill store shelf.

Discovery IA only — not projected onto the prompt tree, not a kind tab.
Persistence leaf: CHECK constraint + API/runtime share the same tuple.
"""

from __future__ import annotations

from typing import Literal

SKILL_STORE_GROUPS = (
    "legal",
    "writing",
    "research",
    "product",
    "engineering",
    "decision",
)

SkillStoreGroupName = Literal[
    "legal",
    "writing",
    "research",
    "product",
    "engineering",
    "decision",
]

SKILL_STORE_GROUP_SQL = (
    "shelf_group in ("
    "'legal', 'writing', 'research', 'product', 'engineering', 'decision'"
    ")"
)


def empty_group_counts() -> dict[str, int]:
    return {name: 0 for name in SKILL_STORE_GROUPS}


def is_skill_store_group(value: str | None) -> bool:
    return value in SKILL_STORE_GROUPS
